kratos/utilities/math_utils.h

```diff
@@ -726,6 +740,39 @@ class KRATOS_API(KRATOS_CORE) MathUtils
         }
     }
 
+    /**
+     * @brief Clamps a value between a minimum and maximum range with a scenario.
+     * @details This function ensures that the given value `rX` lies within the specified
+     * range `[rMinimum, rMaximum]`. If `rX` is less than `rMinimum`, it returns
+     * `rMinimum`. If `rX` is greater than `rMaximum`, it returns `rMaximum`.
+     * Otherwise, it returns `rX`. It also returns a scenario indicating if the value is below the minimum, above the maximum or within the bounds.
+     * @tparam T The type of the value and bounds, typically a numeric type.
+     * @param rX The value to clamp.
+     * @param rMinimum The minimum bound.
+     * @param rMaximum The maximum bound.
+     * @param rScenario The scenario where the value is clamped. It can be BELOW_MINIMUM, ABOVE_MAXIMUM or WITHIN_BOUNDS.
+     * @return The clamped value.
+     */
+    template <typename T>
+    static inline T Clamp(
```

Comment:
```suggestion
    static T Clamp(
```

Reply:
Why remove inline?

Reply:
Because it's unnecessary here. https://en.cppreference.com/w/cpp/language/inline

First of all, the `inline` keyword has **absolutely nothing** to do with compiler inlining. They're completely different things.

The `inline` keyword lets the compiler know that the same definition of a function may be present in multiple translation units. It's basically a tool for allowing function definitions in headers. However, function templates are always implicitly inline since every translation unit has to instantiate them separately, so there's no point in manually adding an `inline` specifier to them.

Reply:
Okay